mobile/src/main/java/org/openhab/habdroid/background/tiles/AbstractTileService.kt

```diff
@@ -25,23 +25,53 @@ import android.service.quicksettings.TileService
 import android.util.Log
 import androidx.annotation.DrawableRes
 import androidx.annotation.RequiresApi
+import androidx.annotation.StringRes
 import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
 import kotlinx.android.parcel.Parcelize
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import org.json.JSONException
 import org.json.JSONObject
 import org.openhab.habdroid.R
 import org.openhab.habdroid.background.BackgroundTasksManager
+import org.openhab.habdroid.background.ItemUpdateWorker
 import org.openhab.habdroid.background.tiles.AbstractTileService.Companion.getPrefKeyForId
 import org.openhab.habdroid.util.getPrefs
 import org.openhab.habdroid.util.showToast
 
 @RequiresApi(Build.VERSION_CODES.N)
 abstract class AbstractTileService : TileService() {
-    @VisibleForTesting abstract val ID: Int
+    @Suppress("PropertyName") @VisibleForTesting abstract val ID: Int
+    var subtitleUpdateJob: Job? = null
 
     override fun onStartListening() {
         Log.d(TAG, "onStartListening()")
         qsTile?.let { updateTile(it) }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+            val workManager = WorkManager.getInstance(applicationContext)
+            val lifeCycleOwner = object : LifecycleOwner {
+                private val lifecycleRegistry = LifecycleRegistry(this).apply {
+                    handleLifecycleEvent(Lifecycle.Event.ON_START)
```

Comment:
I don't think this is sufficient, since you never emit a destroy event. See here for a reference to a full implementation: https://stackoverflow.com/a/50279360

Reply:
You're right, but the issue is, that `onStartListening()` and `onStopListening()` aren't called as they should IMO.
In the log I can see, that Android calls `onStartListening()`, `onClick()` and `onStopListening()` with little time between whenever I click on the tile.

From the docs: "The lifecycle of a TileService is different from some other services in that it may be unbound during parts of its lifecycle."
To me this sounds like the OS will kill the tile service anyway. However I could stop `lifecycleRegistry` when the "Success" message is cleared and start it again in `onClick()`.

Reply:
> To me this sounds like the OS will kill the tile service anyway.

Yes, and the point I was trying to make was to emit a destroy event when that happens (and onDestroy is called).

> However I could stop lifecycleRegistry when the "Success" message is cleared and start it again in onClick().

:+1: